pym/bob/builder.py

```diff
@@ -1265,7 +1261,7 @@ async def _cookCheckoutStep(self, checkoutStep, depth):
         # Generate audit trail before setResultHash() to force re-generation if
         # the audit trail fails.
         if checkoutHash != oldCheckoutHash or self.__force:
-            await self._generateAudit(checkoutStep, depth, checkoutHash, checkoutExecuted)
+            await self._generateAudit(checkoutStep, depth, checkoutHash, checkoutHash, checkoutExecuted)
```

Comment:
The implicit identity of `checkoutHash` with the build ID of the checkout step is quite confusing, if one compares the invocation of `_generateAudit()` within the three functions `_cookCheckoutStep()`, `_cookBuildStep()` and `_cookPackageStep()`. From my point of view this should be made explicit by introducing a variable `checkoutBuildId` consistently with `buildBuildId` and `packageBuildId`, which is set to `checkoutHash`. This makes the code more readable. Maybe this identity could be commented in the code.

Reply:
Good suggestion! Adapted accordingly...